src/project/ProjectModel.js

```diff
@@ -85,6 +85,16 @@ define(function (require, exports, module) {
 
     var _illegalFilenamesRegEx = /^(\.+|com[1-9]|lpt[1-9]|nul|con|prn|aux|)$|\.+$/i;
 
+    /**
+     * CDO-Bramble: We have more strict requirements for allowed filename characters
+     * than Bramble. Use an allowlist instead of a list of invalid characters.
+     */
+    var _validCharsRegex = /[^0-9A-Za-z!\-_.*'()]/;
+    var _validChars = "0-9 a-z A-Z ! - _ . * ' ( )"; // CDO-Bramble: Only for use in error messages.
+    function isValidCdoFilename(filename) {
+        return !(_validCharsRegex.test(filename) || _illegalFilenamesRegEx.test(filename));
```

Comment:
do we still need the illegal regex test?
Also maybe I'm misunderstanding something, but why is it `!(valid || illegal)`? Wouldn't that just return true when neither regex test finds a match?

Reply:
we do still need `_illegalFilenamesRegEx` because it checks for things like making sure a filename doesn't end in a period, which our new regex won't catch.

you aren't misunderstanding! i accidentally named my variable very horribly 😅 it should be `_invalidCharsRegex`, not `_validCharsRegex` -- fixing that now